### PR TITLE
HBASE-28928 Handle NPE in Split/Merge table when getMasterQuotaManager returns null

### PR DESCRIPTION
**Pull request by mnpoonia**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 29s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 49s |  |  master passed  |
| +1 :green_heart: |  compile  |   3m 13s |  |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 45s |  |  master passed  |
| -1 :x: |  spotbugs  |   1m 57s | [/branch-spotbugs-hbase-server-warnings.html](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6375/1/artifact/yetus-general-check/output/branch-spotbugs-hbase-server-warnings.html) |  hbase-server in master has 1 extant spotbugs warnings.  |
| +1 :green_heart: |  spotless  |   0m 53s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 28s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   3m 32s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   3m 32s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 40s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   1m 58s |  |  the patch passed  |
| +1 :green_heart: |  hadoopcheck  |  10m 46s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| +1 :green_heart: |  spotless  |   0m 42s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 10s |  |  The patch does not generate ASF License warnings.  |
|  |   |  39m  9s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6375/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6375 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless |
| uname | Linux 88fb3747b398 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 726b2d8e2f2ba70bedd277bb9f2cc9e2b1749eb9 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| Max. process+thread count | 84 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6375/1/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by mnpoonia** [comment]
Spotbugs failure is unrelated.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/MergeTableRegionsProcedure.java [hunk 1]
```diff
@@ -540,7 +541,10 @@ private void preMergeRegions(final MasterProcedureEnv env) throws IOException {
     }
     // TODO: Clean up split and merge. Currently all over the place.
     try {
-      env.getMasterServices().getMasterQuotaManager().onRegionMerged(this.mergedRegion);
+      MasterQuotaManager masterQuotaManager = env.getMasterServices().getMasterQuotaManager();
```
Comment (NihalJain) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/MergeTableRegionsProcedure.java:544]: Hi @mnpoonia in which case are you seeing this NPE? Master is starting up and quotas has not yet initialized?

Comment (mnpoonia) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/MergeTableRegionsProcedure.java:544]: @NihalJain You are right. A Master was in transition when we got this exception. It was trying to become active. Just after this NPE i can see a log line 

```
2024-10-14 20:02:11,265 ERROR [PEWorker-31] procedure2.ProcedureExecutor - CODE-BUG: Uncaught runtime exception: pid=58745238, state=RUNNABLE:SPLIT_TABLE_REGION_PRE_OPERATION, locked=true; SplitTableRegionProcedure table=Table1, parent=36dcdeb9044bd0ece2eb046cf7fd7c2f, daughterA=cfba8d001ef9d3640060a4e845d3d013, daughterB=0becd69b7ccdc7e12dcc96cc637dc300
java.lang.NullPointerException

2024-10-14 20:02:11,269 INFO  [aster/hmaster-1:61000:becomeActiveMaster] quotas.MasterQuotaManager - Quota support disabled
```

**Review by NihalJain — APPROVED** [review]

**Comment by NihalJain** [comment]
+1, LGTM

**Comment by mnpoonia** [comment]
@NihalJain if everything is good. can you please help me in getting this committed upto branch-2.5.

**Pull request merged**